keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/credential/NTHashPasswordHashProvider.java

```
@@ -0,0 +1,59 @@
+package io.cloudtrust.keycloak.credential;
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.keycloak.credential.hash.PasswordHashProvider;
+import org.keycloak.models.PasswordPolicy;
+import org.keycloak.models.credential.PasswordCredentialModel;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.Security;
+import java.util.HexFormat;
+
+public class NTHashPasswordHashProvider implements PasswordHashProvider {
+    static {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    @Override
+    public boolean policyCheck(PasswordPolicy policy, PasswordCredentialModel credential) {
+        return policy != null && credential.getPasswordCredentialData().getAlgorithm().equals(NTHashPasswordHashProviderFactory.ID);
+    }
+
+    @Override
+    public PasswordCredentialModel encodedCredential(String rawPassword, int iterations) {
+        try {
+            String ntHash = computeNTHash(rawPassword);
+            return PasswordCredentialModel.createFromValues(NTHashPasswordHashProviderFactory.ID, null, iterations, ntHash);
+        } catch (NoSuchAlgorithmException | NoSuchProviderException e) {
+            return null;
```

Please add log message here. We do not want to silently fail.